CORE-344: Add endpoint for hd case diary

marcoebbinghaus opened the pull request:
Added endpoint to get a diary / list of diary entries for a given hd case.

marcoebbinghaus commented:
Thank you, I added the diary link but didn't add the trailing slashes according to our new formatting rules which changed contemporaneous to this PR.

jekutzsche commented on backend/src/main/java/quarano/department/web/TrackedCaseDiaryEntrySummary.java:
```diff
@@ -0,0 +1,80 @@
+public class TrackedCaseDiaryEntrySummary extends RepresentationModel<TrackedCaseDiaryEntrySummary> {
+
+	private final DiaryEntry entry;
+	private final MessageSourceAccessor messages;
```
`Messages` isn't used! Shouldn't the dependence then be better removed?

jekutzsche commented on backend/src/main/java/quarano/department/web/TrackedCaseDiaryEntrySummary.java:
```diff
@@ -0,0 +1,80 @@
+	private final MessageSourceAccessor messages;
+	private final MapperWrapper mapper;
+
+	public TrackedCaseDiaryEntrySummary(DiaryEntry diaryEntry, MessageSourceAccessor messages, MapperWrapper mapper) {
+		this.entry = diaryEntry;
+		this.messages = messages;
+		this.mapper = mapper;
+	}
```
Why do you use an explicit constructor here and not `@RequiredArgsConstructor` like below at `ContactSummary`?

odrotbohm commented on backend/src/main/java/quarano/department/web/TrackedCaseController.java:
```diff
@@ -327,14 +342,22 @@
 				.build();
 	}
 
-	private Stream<TrackedCaseContactSummary> createSummaries(TrackedCase trackedCase) {
```
Let's avoid unrelated renames.

jekutzsche replied:
What do you mean? 
I think the renaming is good to make the sense of the method more concrete compared to the new method createDiaryEntrySummaries.

marcoebbinghaus replied:
Yes, could you please clarify? I don't get what you mean with unrelated names.

odrotbohm replied:
Ahh, that makes a lot of sense then. The context wasn't quite obvious to me. Thanks for elaborating.

marcoebbinghaus commented:
I think you are right and there is no reason for adding the constructor manually here. So I removed the constructor and added the Lombok annotation for that. Also I removed the unused messages field in TrackedCaseDiaryEntrySummary.

odrotbohm commented:
Ready to take it from here. Would you mind squashing the commits into one and brush up the commit message elaborating a bit on the details of what's introduced here? I don't care about the individual commit steps but would rather be able to read a summary of the exposed API, important aspects of the implementation etc.

odrotbohm commented:
That's polished and merged. Thanks!
